code/game/gamemodes/umbrage/umbrage_datum.dm

```diff
@@ -0,0 +1,131 @@
+//Handles everything related to psi and lucidity.
+/datum/umbrage
+	var/datum/mind/linked_mind
+	var/mob/living/linked_body
+	var/psi = 100 //Our psi, used for abilities.
+	var/max_psi = 100
+	var/psi_regeneration = 20 //The limit for psi regenerated during the cycle.
+	var/psi_used_since_last_cycle //How much psi we've used in the last five seconds.
+	var/cycle_progress = 0 //When this reaches 5, it will reset to 0 and regenerate up to 20 spent psi.
+	var/lucidity = 3 //Lucidiy is used to buy abilities. We gain one each time we drain someone.
+	var/lucidity_drained = 3 //How much lucidity we've taken overall.
+	var/list/ability_types = list() //Types of abilities that the umbrage CAN OWN
```

Comment:
This is needed because this is an assoc list, and won't work if it's not declared